hdeps/tests/compatibility.py

```diff
@@ -0,0 +1,50 @@
+import unittest
+from typing import Dict
+from unittest.mock import Mock
+
+from packaging.version import Version
+
+from ..compatibility import requires_python_match
+
+
+class CompatibilityTest(unittest.TestCase):
+    def test_requires_python_match(self) -> None:
+        # pretend the env marker says 3.7
+        python_version = Version("3.7")
+        cache: Dict[str, bool] = {}
+        fake_project = Mock(
+            versions={
+                Version("3.6"): Mock(requires_python="zzz"),
+                Version("3.7"): Mock(requires_python=">=3.7"),
+                Version("3.8"): Mock(requires_python=">=3.8"),
+            }
+        )
+        # This is repeated to test that we read the cached value correctly.
+        self.assertFalse(
+            requires_python_match(fake_project, cache, python_version, Version("3.8"))
+        )
+        self.assertEqual(False, cache[">=3.8"])
```

Comment:
Super minor, but I find that pytest does a better job showing the test assertion failure with a simple:
```
assert not cache[">=3.8"]
```
or to insist:
```
assert cache[">=3.8"] is False
```

Haven't seen the old `self.assertFoos()` in a long time :)
And unless something changed recently on unittest side, the output of those was quite miserable, compared to pytest's out-of-the box showing of why an `assert` failed

Reply:
They're what I can do in stdlib :)

pytest is kind of an all-or-nothing deal, where it's really hard to integrate the tests into other runners.  unittest is the lowest-common-denomimator that keeps maximum compatibility.